projects/core/src/auth/user-auth/facade/auth.service.ts

```diff
@@ -43,13 +43,15 @@ export class AuthService {
    * Check params in url and if there is an code/token then try to login with those.
    */
   async checkOAuthParamsInUrl(): Promise<void> {
-    try {
-      const tokenReceived = getLastValueSync(
-        this.oAuthLibWrapperService.events$.pipe(
-          filter((e) => e.type === 'token_received')
-        )
-      );
+    let tokenReceived: OAuthEvent | undefined;
+    const sub: Subscription = this.oAuthLibWrapperService.events$
+      .pipe(
+        take(1),
+        filter((e: OAuthEvent) => e.type === 'token_received')
+      )
+      .subscribe((e: OAuthEvent) => (tokenReceived = e));
 
+    try {
       const result = await this.oAuthLibWrapperService.tryLogin();
```

Comment:
What do you think about adding a code comment explaining the connection between `tryLogin()` and the event `token_received`?  IMHO the connection is not obvious. It is known only from the source code of the OAuthLib.
```ts

// The method `oAuthLibWrapperService.tryLogin()` obtains the token either from the URL params
// or from the storage. To distinguish those 2 cases, we observe the event `token_received`.
//
// The event 'token_received' is emitted, when the method `oAuthLibWrapperService.tryLogin()`
// can obtain the token from the URL params (which means we've just returned from 
// an external authorization page to Spartacus).
// 
// But the event 'token_received' is not emitted when the method `oAuthLibWrapperService.tryLogin()`
// can obtain the token from the storage (e.g. on refresh of the Spartacus page).
const result = await this.oAuthLibWrapperService.tryLogin();
```

--
Side note:
Ideally, I'd love to encapsulate this implementation detail of the OAuthLib **inside** our `this.oAuthLibWrapperService.tryLogin();`, so the `AuthService` facade is more clean, readable and testable .
This would require changing the signature of `oAuthLibWrapperService.tryLogin()` from `Promise<boolean>` to `Promise<OAuthTryLoginResult>`. (note: the name of the interface is up to discussion)
```ts
interface OAuthTryLoginResult {
   result: boolean,

   /** 
     * It's true only when received the token from URL params.
     * Note: it's false in particular, when the token was obtained from the storage.
     */
   tokenReceived: boolean,
} 
```

This would be a breaking change.
What do you think about merging the current fix just as is to the latest 4.x, but add the improvements in 5.0 (move the complexity of subscriptions to inside tryLogin()) + describe breaking changes in the migration docs?